File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    let arr1 = [];
    let arr2 = [];
    want.forEach((v,i) => arr1.push( ...Array(number[i]).fill(v)) );
    arr1 = arr1.sort().join();
    
    for(let i=0; i<=discount.length-10; i++){
        if(!want.includes(discount[i])) continue;
        arr2 = discount.slice(i,i+10).sort().join();
        
        if(arr1 == arr2){
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;

        for(int i = 0; i < discount.length - 9; i++){
            int[] num = number.clone();
            boolean possible = true;

            for(int j = i; j < i + 10; j++){
                for(int k = 0; k < want.length; k++){
                    if(discount[j].equals(want[k])){
                        num[k]--;
                        break;
                    }
                }
            }

            for(int k = 0; k < num.length; k++){
                if(num[k] != 0){
                    possible = false;
                    break;
                }
            }

            if(possible){
                answer++;
            }
        }

        return answer;
    }
}